js/ad_manager_vast.js

```diff
@@ -442,48 +509,33 @@ OO.Ads.manager(function(_, $) {
      * Also the properties of whether an ad is linear or not, and whether or not the marquee should show are set here.
      * @private
      * @method Vast#addToTimeline
-     * @param {object} ad The ad metadata that is being added to the timeline.
-     * @param {object} adLoaded The ad object that has been loaded.
+     * @param {object} metadata The ad metadata that is being added to the timeline.
+     * @param {boolean} True if the ad was added to the timeline successfully, false otherwise
      */
-    var addToTimeline = _.bind(function(ad, adLoaded) {
-      if (!ad) return;
+    var addToTimeline = _.bind(function(metadata) {
+      if (!metadata) return false;
       var timeline = [];
       var type, duration;
 
-      if (!_.isEmpty(ad.data.linear.mediaFiles)) {
+      if (!_.isEmpty(metadata.data.linear.mediaFiles)) {
         type = this.amc.ADTYPE.LINEAR_VIDEO;
-        duration = OO.timeStringToSeconds(ad.data.linear.Duration);
+        duration = OO.timeStringToSeconds(metadata.data.linear.Duration);
       }
       else
       {
         type = this.amc.ADTYPE.NONLINEAR_OVERLAY;
-        duration = ad.data.nonLinear.Duration ?  OO.timeStringToSeconds(ad.data.nonLinear.Duration) : 0;
+        duration = metadata.data.nonLinear.Duration ?  OO.timeStringToSeconds(metadata.data.nonLinear.Duration) : 0;
       }
-      var positionSeconds = adLoaded.time/1000;
 
-      // Save the stream data for use by VideoController
-      var streams = {};
-      var linearAd = ad.data.linear;
```

Comment:
Why was this removed? We need to build the streamData object ot have all the supported streams listed so the VTC can pick the right one.